server/src/main/java/org/opensearch/action/support/clustermanager/term/TransportGetTermVersionAction.java

```diff
@@ -76,7 +76,7 @@ protected void clusterManagerOperation(
         ClusterState state,
         ActionListener<GetTermVersionResponse> listener
     ) throws Exception {
-        ActionListener.completeWith(listener, () -> buildResponse(request, state));
+        ActionListener.completeWith(listener, () -> buildResponse(request, clusterService.preCommitState()));
```

Comment:
Lets control this with a setting. The implication is we are returning a state for which appliers are not run yet. There can be a rare case where latest state is reported using _cluster/state API before running applier can cause some issue though in the core this shouldnt be an issue as core process would never call _cluster/state.